### PR TITLE
Update train.py

### PR DESCRIPTION
**Pull request by francoep**
removed offset, it doesn't do anything
Added feature --percent_reduced X which will evaluate on a reduced subsample of X% of the provided types file during training for faster performance of large files

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -315,41 +314,59 @@ def update_from_result(name, test, result):
...
+        counter+=1
+        test_idxs['test2']=counter
+        if use_reduced:
             test_models += ['trainreducedtest2.%d.prototxt' % pid]
             test_files += [files['reduced_test2']]
             test_roots += [args.data_root2]
-    if not test_on_train:
+            counter+=1
+            test_idxs['reduced_test2']=counter
+    if not test_on_train:#TODO - check if this works as intended
```
Comment (dkoes) [inline train.py]: well, does it?

File: train.py [hunk 2]
```diff
@@ -427,11 +444,14 @@ def update_from_result(name, test, result):
...
         solver.net.copy_from(args.weights) #TODO this doesn't actually set the necessary weights...
-                
+
     test_nets = {}
     for key, test_file in files.items():
-        idx = test_files.index(test_file)
-        test_nets[key] = solver.test_nets[idx], count_lines(test_file), 0
+        idx = test_idxs[key]
+        if args.percent_reduced and 'reduced' in key:
+            test_nets[key] = solver.test_nets[idx], int(count_lines(test_file)*args.percent_reduced/100)
```
Comment (dkoes) [inline train.py]: take max of int(count_lines...) and 1

File: train.py [hunk 3]
```diff
@@ -768,6 +792,8 @@ def get_train_test_files(prefix, foldnums, allfolds, reduced, prefix2):
         i = 'all'
         files[i] = {}
         files[i]['train'] = files[i]['test'] = '%s.types' % prefix
+        if bool(percent_reduced):
+            files[i]['reduced_train'] = files[i]['reduced_test'] = '%s.types' % prefix
         if reduced:
```
Comment (dkoes) [inline train.py]: should be elif?

File: train.py [hunk 4]
```diff
@@ -785,17 +811,28 @@ def get_train_test_files(prefix, foldnums, allfolds, reduced, prefix2):
 
     #identify all train/test pairs
     try:
-        train_test_files = get_train_test_files(args.prefix, args.foldnums, args.allfolds, args.reduced, args.prefix2)
+        train_test_files = get_train_test_files(args.prefix, args.foldnums, args.allfolds, args.reduced, args.prefix2, args.percent_reduced)
     except OSError as e:
         print "error: %s" % e
         sys.exit(1)
 
+    print 'EASYFIND',train_test_files
```
Comment (dkoes) [inline train.py]: print statement

**Review by dkoes — CHANGES_REQUESTED** [review]
minor comments

**Comment by dkoes** [comment]
Why are their two pull requests for this commit?

**Comment by francoep** [comment]
The pull request for the other file was me just deleting ", _" from 1 line of the code (and is sure to be accepted.

File: train.py (hunk 3 above)
Comment (francoep) [reply train.py]: Once this is called it has already passed the check that only one of bool(percent_reduced) and reduced is set to True, and I wanted to keep the structure of the code as separate if statements intact. The separate if statements and an elif would output the same result.

File: train.py (hunk 1 above)
Comment (francoep) [reply train.py]: yes (forgot to delete the TODO)

**Review by dkoes — APPROVED** [review]

**Pull request merged**